osquery/tables/system/linux/disk_encryption.cpp

```diff
@@ -114,34 +148,40 @@ QueryData genFDEStatus(QueryContext& context) {
     return results;
   }
 
-  std::map<std::string, Row> encrypted_rows;
-
   bool runSelectAll(true);
-  QueryData block_devices;
+  std::vector<std::string> queried_devices;
+  std::map<std::string, Row> block_devices;
+  std::map<std::string, Row> encrypted_rows;
```

Comment:
```suggestion
  // Track the devices in the context, so that if we query the parent encryption, we don't return that data. 
  // sqlite would filter this, but it seems cleaner here. 
  std::vector<std::string> queried_devices;

  // Because this table works by querying the `block_devices` table, cache those results to avoid O(N^2) issues
  std::map<std::string, Row> block_devices;

  // Cache what devices are encrypted. This allows us to reuse the calculations. This is important for determining parent encryption status. 
  std::map<std::string, Row> encrypted_rows;
```